beluga_amcl/src/ndt_amcl_node_3d.cpp

```diff
@@ -152,6 +154,18 @@ NdtAmclNode3D::NdtAmclNode3D(const rclcpp::NodeOptions& options) : BaseAMCLNode{
 
 void NdtAmclNode3D::do_activate(const rclcpp_lifecycle::State&) {
   RCLCPP_INFO(get_logger(), "Making particle filter");
+
+  map_visualization_pub_->on_activate();
+
+  // Publish markers for map visualization
+  beluga_ros::msg::MarkerArray obstacle_markers{};
+  bool flag;
+  beluga_ros::assign_obstacle_map(map_, obstacle_markers, flag);
+  if (flag) {
+    RCLCPP_WARN(get_logger(), "Covariances from map representation cells seem to be non-diagonalizable");
```

Comment:
What happens in this case, are some of the markers not generated? Please explain that in the warning message.

Reply:
in that case, we create some cell-size cubes instead of generate an ellipsoid based on the covariance.
the color is different to indicate the error, and in should not happen in a well generated map (it was not a problem in the demo we have)

Reply:
+1 to the name of the variable.
It may be better to return a `std::pair<MarkerArray, bool>` as well.

Reply:
Thanks for the context! Sounds reasonable.

Suggestion:

```cpp
RCLCPP_WARN(get_logger(), "Some cell covariances appear to be non-diagonalizable. A cube will be generated instead of an ellipsoid for those.");
```